src/hooks/useSidePanel.js

```diff
@@ -0,0 +1,29 @@
+import { useCallback, useState } from 'react'
+
+export function useSidePanel() {
+  const [visible, setVisible] = useState(false)
+  const [opened, setOpened] = useState(false)
+
+  const requestOpen = useCallback(() => {
+    setVisible(true)
+    setOpened(false)
+  }, [setVisible, setOpened])
```

Comment:
This is not needed, `useState()` setters are guaranteed to never change.
```suggestion
  }, [])
```

Reply:
yeah, good point